UPGRADE_NOTES.md

```diff
@@ -1,4 +1,55 @@
-### Upgrading from 0.14 to 0.15
+## Upgrading from 0.16 to 0.17
+
+- Ensure you've upgraded to crystal 0.30.0
+- Upgrade Lucky CLI (homebrew)
+
+```
+brew update
+brew upgrade crystal-lang # Make sure you're up-to-date. Requires 0.30.0
+brew upgrade lucky
+```
+
+- Upgrade Lucky CLI (Linux)
+
+- Update `.crystal-version` to `0.30.0`
+
+> Remove the existing Lucky binary and follow the Linux
+> instructions in this section
+> https://luckyframework.org/guides/getting-started/installing#on-linux
+
+- Update versions in `shard.yml`
+  - Lucky should be `~> 0.17`
+- Run `shards update`
+
+### General updates
+- Rename: Action rendering method `text` to `plain_text`.
+- Update: use of `number_to_currency` now returns a String instead of writing to the view directly.
+- Rename: `Avram::Repo` to `Avram::Database` in `config/database.cr`.
+
+### Updating queries
+- Rename: `Query.destroy_all` to `Query.truncate`.
+- Rename: all association query methods from the association name to `where_{association_name}`. (e.g. `UserQuery.new.posts` => `UserQuery.new.where_posts`)
+- Update: all association query methods no longer take a block. Pass the query in as an argument. (e.g. `UserQuery.new.posts { |post_query| }` => `UserQuery.new.where_posts(PostQuery.new)`)
+
+### Moving forms to operations
+- Rename: the `src/forms` directory to `src/operations`.
+- Rename: all `BaseForm` mentions to `SaveOperation` in `src/operations`. (e.g. `User::BaseForm` => `User::SaveOperation`)
+- Rename: `fillable` to `permit_columns`
+- Rename: form class names to new naming convention. (e.g. `class UserForm < User::SaveOperation` => `class SaveUser < User::SaveOperation`). This step is optional, but still recommended to avoid future confusion.
+- Rename: `Avram::VirtualForm` to `Avram::Operation`.
+- Rename: virtual form class names to new naming convention. (e.g. `class SignInForm < Avram::Operation` => `class SignInUser < Avram::Operation`).
+- Rename: `virtual` to `attribute`.
+- Update: all `SaveOperation` classes to call `before prepare`. The `prepare` method is no longer called by default, which allows you to rename this method as well. Also note that validations are no longer called by default, and any validations not in your current `prepare` method should be added in a `before do/end` block.
+- Rename: use of field `form_name` to attribute `param_key`.
```

Comment:
An example would be helpful here as well. Might be good to mention this may not even require a change. It is only necessary if you are using the `param_key` macro in an operation.

Also maybe `use of field` -> "the `form_name` macro"